src/components/Form/FormField.tsx

```diff
@@ -0,0 +1,221 @@
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+
+import { UIComponent, customPropTypes, childrenExist, createShorthandFactory } from '../../lib'
+import { ComponentVariablesInput, ComponentPartStyle } from '../../../types/theme'
+import {
+  Extendable,
+  ReactChildren,
+  ShorthandValue,
+  ShorthandRenderFunction,
+} from '../../../types/utils'
+import Text from '../Text'
+import { createSlotFactory } from '../Slot/Slot'
+
+export interface IFormFieldProps {
+  as?: any
+  children?: ReactChildren
+  className?: string
+  label?: ShorthandValue
+  renderLabel?: ShorthandRenderFunction
+  controlType?: React.ReactType<any>
+  control?: ShorthandValue
+  renderControl?: ShorthandRenderFunction
+  inline?: boolean
+  message?: ShorthandValue
+  renderMessage?: ShorthandRenderFunction
+  required?: boolean
+  styles?: ComponentPartStyle
+  variables?: ComponentVariablesInput
+}
+
+/**
+ * A field is a form element containing a label and an input.
+ */
+class FormField extends UIComponent<Extendable<IFormFieldProps>, any> {
+  public static displayName = 'FormField'
+
+  public static className = 'ui-form__field'
+
+  static create: Function
+
+  public static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /**
+     *  Button content for childrenApi
+     *  @docSiteIgnore
+     */
+    children: PropTypes.node,
+
+    /** Additional CSS class name(s) to apply.  */
+    className: PropTypes.string,
+
+    /**
+     * A form control component (i.e. Input) or HTML tagName (i.e. 'input').
+     * Extra FormField props are passed to the control component.
+     * Mutually exclusive with children.
+     */
+    controlType: customPropTypes.some([
+      PropTypes.func,
+      PropTypes.oneOf(['button', 'input', 'select', 'textarea']),
```

Comment:
can we use these to define the `controlType ` prop in the interface?

Reply:
We don't know all components that can be defined there upfront, that is why the type is `React.ReactType<any>`. We know just for the primitive types that can be one of those.

Reply:
NA anymore